data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/StrictResponseEventHandlingStrategy.java

```diff
@@ -19,8 +20,10 @@ public class StrictResponseEventHandlingStrategy implements ResponseEventHandlin
     public List<Record<Event>> handleEvents(List<Event> parsedEvents,
                                             List<Record<Event>> originalRecords) {
         if (parsedEvents.size() != originalRecords.size()) {
-            throw new RuntimeException(
-                    "Response Processing Mode is configured as Strict mode but behavior is aggregate mode. Event count mismatch.");
+            throw new StrictResponseModeNotRespectedException(
+                    "Response Processing Mode is configured as Strict mode but behavior is aggregate mode. " +
```

Comment:
"The aws_lambda processor is configured with response_events_match set to true. The Lambda function responded with a different number of events. Either set response_events_match to false or investigate your Lambda function to ensure that it returns the same number of events and provided as input. parsedEvents size = . Original events size = "